pkg/prometheus/promcfg.go

```diff
@@ -873,6 +873,7 @@ func (cg *ConfigGenerator) generatePodMonitorConfig(
 
 	//nolint:staticcheck // Ignore SA1019 this field is marked as deprecated.
 	if ep.BearerTokenSecret.Name != "" {
+		level.Warn(cg.logger).Log("msg", "'bearerTokenSecret' is deprecated, use 'authorization' instead.", "podmonitor", m.Name, "namespace", m.Namespace)
```

Comment:
the logger should already have the `pod_monitor="xxx/xxx"` keyval.
```suggestion
		level.Warn(cg.logger).Log("msg", "'bearerTokenSecret' is deprecated, use 'authorization' instead.")
```

Reply:
Oh, maybe it can be added as a keyval to the logger at the top of the function instead, i'll check how.
I will also add a warning to the  other deprecations

Reply:
Oh yes you're right that it should but it's not showing in my logs for some reason

Reply:
So the weird part is that when i use logfmt, I have the following logs without the resource information

```
level=warn ts=2024-03-20T20:26:27.127226961Z caller=promcfg.go:819 component=prometheus-controller msg="'targetPort' is deprecated, use 'port' instead." version=v2.50.1
level=warn ts=2024-03-20T20:26:27.127415876Z caller=promcfg.go:819 component=prometheus-controller msg="'targetPort' is deprecated, use 'port' instead." version=v2.50.1
```

But when I use json is it there:

```
{"[pod_monitor crossplane/crossplane]":"(MISSING)","caller":"promcfg.go:819","component":"prometheus-controller","level":"warn","msg":"'targetPort' is deprecated, use 'port' instead.","ts":"2024-03-20T20:25:44.813625012Z","version":"v2.50.1"}
```

Reply:
Oh i found the issue

Reply:
I ran this little script:

```go
func WithKeyVals(logger log.Logger, keyvals ...interface{}) log.Logger {
	return log.WithSuffix(logger, keyvals...)
}

func main() {
	logger := log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
	logger = WithKeyVals(logger, "caller", log.DefaultCaller)
	level.Info(logger).Log("msg", "hello")
	level.Warn(logger).Log("msg", "hello")
}
```

And we do not add the spread operator (...) after keyvals in the WithKeyVals function, the we pass the keyval function parameter as an array hence this in json: "[pod_monitor crossplane/crossplane]":"(MISSING)"